mission-control-client/src/components/dashboard/admin-dashboard/ProductList.js

```diff
@@ -6,22 +6,28 @@ import { Input } from 'semantic-ui-react'
 
 import Product from "./Product";
 
-// testing with dummy data for products
-import res from "../../../data/projects";
+const ProductList = props => {
+  
+  useEffect(() => {
+    setFiltered({ products: props.products });
+  }, [props]);
 
-const ProductList = () => {
-  // This data will most likely be passed down as props from DashboardHome
-  const [filtered, setFiltered] = useState({ items: res.data.projects });
+  const [filtered, setFiltered] = useState({ products: [] });
 
   const handleChange = e => {
-    if (e.target.value !== "" && filtered.items.length) {
-      let items = filtered.items;
-      items = items.filter(item => {
-        return item.name.toLowerCase().search(e.target.value.toLowerCase()) !== -1;
+    const products = props.products;
+    const re = /^[a-z0-9]+$/i;
+
+    if (e.target.value !== "" && re.test(e.target.value)) {
+      setFiltered({
+        products: products.filter(item => {
+          return item.productName.toLowerCase().search(e.target.value.toLowerCase()) !== -1;
+        })
```

Comment:
Are parts/all of this maybe worth extracting as a util? Would make testing a lot easier.